packages/core-data/src/resolvers.js

```diff
@@ -165,8 +165,44 @@ export const getEntityRecord =
 					}
 				}
 
-				const record = await apiFetch( { path } );
-				dispatch.receiveEntityRecords( kind, name, record, query );
+				const response = await apiFetch( { path, parse: false } );
+				const record = await response.json();
+
+				const allowHeader = response.headers?.get( 'allow' );
+				const allowedMethods = allowHeader?.allow || allowHeader || '';
+				const permissions = {};
+				const methods = {
+					create: 'POST',
+					read: 'GET',
+					update: 'PUT',
+					delete: 'DELETE',
+				};
+				for ( const [ actionName, methodName ] of Object.entries(
+					methods
+				) ) {
+					permissions[ actionName ] =
+						allowedMethods.includes( methodName );
+				}
+
+				registry.batch( () => {
+					dispatch.receiveEntityRecords( kind, name, record, query );
+
+					for ( const action of [
+						'create',
+						'read',
+						'update',
+						'delete',
```

Comment:
Those will be redeclared more than once, too. Should we deduplicate?

Reply:
+1. I will extract this into a constant.